### PR TITLE
My awesome work with networking

### PR DESCRIPTION
**Pull request by puptup**

### CODE REVIEW

File: network1/hm1/main.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	Headers    headers `json:"headers"`
+}
+
+type headers struct {
+	Accept    []string `json:"Accept"`
+	UserAgent []string `json:"User-Agent"`
+}
+
+func main() {
+	req := &request{
```
Comment (wshaman) [inline network1/hm1/main.go:22]: This data should be grabbed from `http.Request` and display real UserAgent, headders,... etc

File: network1/hm2/client.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+	conn, err := net.Dial("tcp", "127.0.0.1:8081")
+	onErrPanic(err)
+
+	for {
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Print("Text to send: ")
+		text, err := reader.ReadString('\n')
+		onErrPanic(err)
+		if len(text) == 6 {
```
Comment (wshaman) [inline network1/hm2/client.go:21]: haha. Nice workaround, try `strings.Trim` instead.
EOL symbol could be "\r\n" or "\n" based on OS, and this check will fail.

File: network1/hm2/client.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+			if text[:4] == "exit" {
+				os.Exit(2)
```
Comment (wshaman) [inline network1/hm2/client.go:23]: Why `2`? It actually means `wrong usage`, OK exit should be `0`

File: network1/hm3/server.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+func getPostForm(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/" {
+		http.Error(w, "404 not found.", http.StatusNotFound)
+		return
+	}
+
+	switch r.Method {
+	case "GET":
```
Comment (wshaman) [inline network1/hm3/server.go:16]: it's a good practice to use `http.MethodGet` const to avoid some misspells.

File: network1/hm3/server.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		address := r.FormValue("address")
+		token := name + ":" + address
+		cookie := http.Cookie{
+			Name:  "token",
+			Value: token,
+		}
+		http.SetCookie(w, &cookie)
+		http.ServeFile(w, r, "form.html")
+	default:
+		fmt.Fprintf(w, "Sorry, only GET and POST methods are supported.")
```
Comment (wshaman) [inline network1/hm3/server.go:33]: It should be an HTTP error Method Not Allowed.
https://httpstatuses.com/

File: network1/hm1/main.go (hunk 1 above)
Comment (puptup) [reply network1/hm1/main.go]: I fixed it :)

File: network1/hm1/main.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+	"encoding/json"
+	"log"
+	"net/http"
+)
+
+type request struct {
+	Host       string  `json:"host"`
+	UserAgent  string  `json:"user_agent"`
+	RequestURL string  `json:"request_uri"`
+	Headers    headers `json:"headers"`
```
Comment (wshaman) [inline network1/hm1/main.go:13]: `Headers
```suggestion
	Headers   map[string]string `json:"headers"`
```

Comment (wshaman) [reply network1/hm1/main.go:13]: This will allow to dump all headers

Comment (puptup) [reply network1/hm1/main.go]: But must be map[string][]string, because r.Header returns slice of strings. 
If you meant it, then I did it.

**Pull request merged**